### DISCUSSION TITLE
Bulk download using --link

### DISCUSSION THREAD

**Discussion by etba254** [Serene-Arc/bulk-downloader-for-reddit#450] [Q&A] [answered]
I have a text file where each line is a reddit id:

abcde
fghij
klmno

any way to download all of these straight from the text file, sorta like how youtube-dl works with -a.
Or any suggestions for scripting it to run bdfr on each code individually.

Also, shoutout to Ali Parlakçı, great work.

**Comment by Serene-Arc** [comment, marked as answer]
There is no option to do this through the BDFR itself, however it's fairly trivial to do with common Linux tools. The `--link` option can be supplied multiple times. 

If you wanted to download one link per run, the following would do:

```bash
cat ids.txt | xargs -L 1 python3 -m bdfr download <OPTIONS> --link
```

But it would be more efficient to use multiple links at once. For the that, the following should be sufficient:

```bash
cat links.txt | while read line; do echo "--link" $line; done | xargs python3 -m bdfr download <OPTIONS>
```

If there are a lot of links, then you may run into the maximum line length for any command, so use the `-L` option of xargs to limit it to a smaller number of links at a time.
